application-licensing-licensor/application-licensing-licensor-ui/src/main/resources/Licenses/Code/MissingLicenseMessageMacro.xml

```diff
@@ -0,0 +1,374 @@
+      </name>
+      <priority>
+        <disabled>0</disabled>
+        <name>priority</name>
+        <number>11</number>
+        <numberType>integer</numberType>
+        <prettyName>Priority</prettyName>
+        <size>10</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+## We target the main wiki explicitly because licenses are managed globally.
+#if ($xwiki.exists($services.model.createDocumentReference($xcontext.mainWikiName, 'XWiki', 'ConfigurableClass')))
+  #set ($licenseHomeURL = $xwiki.getURL(
+    $services.model.createDocumentReference($xcontext.mainWikiName, 'XWiki', 'XWikiPreferences'),
+    'admin', 'editor=globaladmin&amp;section=Licenses'))
+#else
+  ## Fall-back in case the Administration Application is not installed.
+  #set ($licenseHomeURL =
+    $xwiki.getURL($services.model.createDocumentReference($xcontext.mainWikiName, 'Licenses', 'WebHome')))
+#end
+#set ($extensionName = $services.localization.render("${wikimacro.parameters.extensionName}"))
+#set ($translation = $services.localization.render("licensor.missingLicense", 'xwiki/2.1',
+  [$extensionName, 'LINKSTART', 'LINKEND']))
+#set ($linkLabel = $stringtool.substringBetween($translation, 'LINKSTART', 'LINKEND'))
+#set ($escapedLinkLabel = $services.rendering.escape($linkLabel, 'xwiki/2.1'))
+#set ($escapedLinkTarget = $services.rendering.escape("path:${licenseHomeURL}", 'xwiki/2.1'))
+{{error}}
+  $translation.replace("LINKSTART${linkLabel}LINKEND", "[[$escapedLinkLabel&gt;&gt;$escapedLinkTarget]]")
+{{/error}}
```

Comment:
This looks more complicated than it should be. Can we deprecate this translation key and add another one that has a single parameter, the link?

Reply:
I updated it. But I couldn't find a best practice for deprecating a translation. So I just deleted the old one and when applications will upgrade the licensing dependency, they have to take care of using the new one.  
What I found was only for platform https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HTranslationPropertyDeprecations , but it does not seem to be of help here